src/main/java/com/google/sps/servlets/PersonalDataServlet.java

```diff
@@ -40,20 +40,22 @@
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get current user email to generate entity key
-    String userEmail = UserServiceFactory.getUserService().getCurrentUser().getEmail();
-    Key userKey = KeyFactory.createKey("User", userEmail);
+    String email = UserServiceFactory.getUserService().getCurrentUser().getEmail();
+    Key userKey = KeyFactory.createKey("User", email);
 
     try {
       // Get current user details from datastore
       Entity entity = DatastoreServiceFactory.getDatastoreService().get(userKey);
       String name = (String) entity.getProperty("name");
-      String department = (String) entity.getProperty("department");
+      String department = (String) entity.getProperty("department");      
+      int year = (int) (long) entity.getProperty("year");
```

Comment:
Why the double cast? `(int) entity.getProperty("year")` should be enough

Reply:
apparently entity.getProperty returns an object and Casting it to int will gives an error but not to long.

Reply:
Looks like Integer is [a supported data type](https://cloud.google.com/datastore/docs/concepts/entities#integer) by datastore. Can you please run `mvn package clean` to delete local datastore entities and see if the error still occurs? (I'm just worried that a `year` is stored there as a `long` that's why casting it as integer thrown an exception.

Reply:
Hi Mostafa, thank you for yor comment, so apparently entity.getProperty("department") returns an object
When casted directly to int, it gives an error
java.lang.ClassCastException: java.lang.Long cannot be cast to java.lang.Integer
However it can be casted to long from Long Object, and once its a long, it can be converted to int.
There are a few ways we can implement to int, one of the easiest is the double cast.
another one is use .intValue() but we have to case the object to Long first to call the fucntion, i.e. ((Long) entity.getProperty("year")).intValue()